metricsbp/prometheus.go

```diff
@@ -0,0 +1,24 @@
+package metricsbp
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var activeRequests = promauto.NewGauge(prometheus.GaugeOpts{
+	Name: "active_requests",
+	Help: "The number of requests being handled by the service.",
+})
```

Comment:
in statsd this is a "runtime gauge" instead of a plain gauge. a "runtime gauge" comes with two tags, [the hostname and the pid, and auto add "runtime." prefix](https://github.com/reddit/baseplate.go/blob/df2b3cb40d2bf1f2e9677300b3e5e265c29ba3ba/metricsbp/runtime_stats.go#L131-L153).

then we have some special handling on telegraf on everything under `runtime.` prefix: we strip the 2 special tags, and calculate the average/max/min/etc. from all the pods reporting this gauge. this is kind of like treating it as a histogram instead of a gauge.

we probably should have an internal spec discussion on how do we want to handle runtime gauges on prometheus first.

Reply:
This doesn't really exist the same way in Prometheus. This is because Prometheus always includes the instance of the monitored target in every metric. For Go, there is only ever one process, so effectively every metric includes those tags.

Reply:
That part of the spec needs to be corrected for Prometheus.